docs/mgmt/readme.md

```diff
@@ -0,0 +1,658 @@
+
+In the design of .Net management SDK, one resource must only have one constant resource type, therefore the generator will check if the resource type which it calculates is a constant, if the resource type is not a constant, it means that the swagger might be using one request path to represent a set of similar resources with different resource types. If this happens, the generator will try to expand the resource types into multiple constant resource types.
+
+For instance, if the above procedure is applied on this request path `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/dnsZones/{zoneName}/{recordType}/{relativeRecordSetName}`, it produces the resource type `Microsoft.Network/dnsZones/{recordType}` which contains a variable in it and is not a constant. If this happens, the generator will try to expand it into multiple constant resource types. This requires the variables in the resulted resource type to be an enum type, if this condition is not met, the generator will throw exceptions for you to resolve by adding configuration/directives.
+
+We have some ways to change the resource type of a resource, please see [change the resource type](#change-the-resource-type) section.
+
+### Singleton resources
+
+Some resources are recognized as singleton resources automatically by the generator, for instance, a request path `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccount/{accountName}/blobServices/default` is a singleton resource with the name `default` and its parent is `StorageAccount`.
+
+The generator will mark a resource as a singleton, when its request path subtracted by the request path of its parent is a constant. For example, the resource with request path `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccount/{accountName}/blobServices/default` has a parent of `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccount/{accountName}`, if you subtract them, you will get `blobServices/default` which is a constant, therefore the resource `BlobService` is a singleton resource.
+
+When a resource is marked as a singleton resource, it will not have its corresponding `[Resource]Collection` class, only have `[Resource]Resource` class and `[Resource]Data` class. The method to get a singleton resource on its parent will be called as `Get[Resource]` in singular form, like
+
+```csharp
+public partial class StorageAccountResource : ArmResource
+{
+    public virtual BlobServiceResource GetBlobService()
+    {
+        /* ... */
+    }
+}
+```
+
+If you want to change the behavior how the generator identifies singleton resources, please see [change singleton resources](#change-singleton-resources) section.
+
+## Management plane configurations
```

Comment:
which file shall I use to store cofiguration?

Reply:
Good question - I will add some general comments here to show a usual way to populate the configurations.
But I kind of suppose we will have another document about how to generate the SDKs to azure-sdk-for-net, and this will be explained there.
The other document about how to generate the SDKs should be the first document that the user of our generator should read first, this document is the second one to read, which is more advanced.
But I will leave a note here about that